src/hotspot/share/services/diagnosticCommand.hpp

```diff
@@ -576,6 +576,30 @@ class CompileQueueDCmd : public DCmd {
   virtual void execute(DCmdSource source, TRAPS);
 };
 
+class VMInspectDCmd: public DCmdWithParser {
+protected:
+  DCmdArgument<char*> _arg1;
+  DCmdArgument<bool> _verbose;
+public:
+  static int num_arguments() { return 1; }
+  VMInspectDCmd(outputStream* output, bool heap);
+  static const char* name() {
+    return "VM.inspect";
+  }
+  static const char* description() {
+    return "Inspect VM object at address.";
```

Comment:
Maybe distinguishing between java heap objects and VM objects would be useful to the reader/user.